File: app/src/main/java/com/yalonso/interview/controllers/ProductDetailsController.java
package com.yalonso.interview.controllers;

import com.yalonso.interview.models.Product;

import java.util.ArrayList;
import java.util.List;

public class ProductDetailsController {
    private static ProductDetailsController productControllerInstance;
    public Product product;

    public static ProductDetailsController getInstance() {
        if (productControllerInstance == null) {
            productControllerInstance = new ProductDetailsController();
        }
        return productControllerInstance;
    }
}

File: app/src/main/java/com/yalonso/interview/controllers/ProductDetailsController.kt
package com.yalonso.interview.controllers

import com.yalonso.interview.models.Product

class ProductDetailsController {
    var product: Product? = null

    companion object {
        @JvmStatic
        private var productControllerInstance: ProductDetailsController? = null
        @JvmStatic
        val instance: ProductDetailsController
            get() {
                if (productControllerInstance == null) {
                    productControllerInstance = ProductDetailsController()
                }
                return productControllerInstance!!
            }
    }
}